### DISCUSSION TITLE
Import `VERCEL_URL` into the SvelteKit client code

### DISCUSSION THREAD

**Discussion by igorlanko** [vercel/vercel#11019] [Ideas]
### Idea

I think Vercel used to expose `VITE_` prefixed variables for `v0`, however it doesn't seem to be supported in `1.x` or `2.x` anymore.

Best I could come up with is keeping empty `VERCEL_` envs locally, and `PUBLIC_APP_` envs in Vercel with a utility like this:
```ts
import { VERCEL_ENV, VERCEL_URL } from '$env/static/private'
import { PUBLIC_APP_ENV, PUBLIC_APP_URL } from '$env/static/public'

export const getAppUrl = () => {
	if (VERCEL_URL) {
		return `https://${VERCEL_URL}`
	} else {
		return PUBLIC_APP_URL
	}
}

export const getAppEnv = () => {
	if (VERCEL_ENV) {
		return VERCEL_ENV
	} else {
		return PUBLIC_APP_ENV
	}
}
```

However as you can see it imports `VERCEL_` from `private`, which is not accessible on the client code. And there's no `VECEL_` exported variables in `public`.

Unless I'm missing something, but it would be nice if `Vercel` started exposing public system variables again?

**Comment by GaryB432** [comment]
without this feature how would one use, e.g., [VERCEL_COMMIT_REF](https://vercel.com/docs/projects/environment-variables/system-environment-variables), in a svelte-kit application?

**Reply by igorlanko** [reply]
As a workaround, you could import `VERCEL_COMMIT_REF` from `static/private` in a server load function, and then return it to the client. Obviously exercise caution doing so.

Clearly this isn't ideal.
